provider/cloudflare/cloudflare.go

```diff
@@ -140,25 +140,71 @@ type RecordParamsTypes interface {
 }
 
 // getUpdateDNSRecordParam is a function that returns the appropriate Record Param based on the cloudFlareChange passed in
-func getUpdateDNSRecordParam(cfc cloudFlareChange) cloudflare.UpdateDNSRecordParams {
-	return cloudflare.UpdateDNSRecordParams{
+func getUpdateDNSRecordParam(cfc cloudFlareChange) (cloudflare.UpdateDNSRecordParams, error) {
+	recordParam := cloudflare.UpdateDNSRecordParams{
 		Name:    cfc.ResourceRecord.Name,
 		TTL:     cfc.ResourceRecord.TTL,
 		Proxied: cfc.ResourceRecord.Proxied,
 		Type:    cfc.ResourceRecord.Type,
-		Content: cfc.ResourceRecord.Content,
 	}
+
+	if cfc.ResourceRecord.Type == "SRV" {
+		recordParam.Data = cfc.ResourceRecord.Data
+	} else {
+		recordParam.Content = cfc.ResourceRecord.Content
+	}
+
+	return recordParam, nil
 }
 
 // getCreateDNSRecordParam is a function that returns the appropriate Record Param based on the cloudFlareChange passed in
-func getCreateDNSRecordParam(cfc cloudFlareChange) cloudflare.CreateDNSRecordParams {
-	return cloudflare.CreateDNSRecordParams{
+func getCreateDNSRecordParam(cfc cloudFlareChange) (cloudflare.CreateDNSRecordParams, error) {
+	recordParam := cloudflare.CreateDNSRecordParams{
 		Name:    cfc.ResourceRecord.Name,
 		TTL:     cfc.ResourceRecord.TTL,
 		Proxied: cfc.ResourceRecord.Proxied,
 		Type:    cfc.ResourceRecord.Type,
-		Content: cfc.ResourceRecord.Content,
 	}
+
+	if cfc.ResourceRecord.Type == "SRV" {
+		recordParam.Data = cfc.ResourceRecord.Data
+	} else {
+		recordParam.Content = cfc.ResourceRecord.Content
+	}
+
+	return recordParam, nil
+}
+
+// parseSRVContent parses the SRV record content string into the structured data required by the Cloudflare API
+func parseSRVContent(content string) (map[string]interface{}, error) {
+	parts := strings.Fields(content)
+	if len(parts) != 4 {
+		return nil, fmt.Errorf("invalid SRV record content: %s", content)
+	}
+
+	priority, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return nil, fmt.Errorf("invalid priority in SRV record content: %s", parts[0])
+	}
+
+	weight, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return nil, fmt.Errorf("invalid weight in SRV record content: %s", parts[1])
+	}
+
+	port, err := strconv.Atoi(parts[2])
+	if err != nil {
+		return nil, fmt.Errorf("invalid port in SRV record content: %s", parts[2])
+	}
+
+	target := parts[3]
```

Comment:
Wdyt of checking IP format with [ParseIP](https://pkg.go.dev/net#ParseIP) ?

Reply:
It wouldn't really make sense here afaik because the the target could be a hostname that would have to be resolved, so pretty much anything goes as long as the string has a `.` in it.